app/services/padlock/token_generator.rb

```diff
@@ -0,0 +1,27 @@
+module Padlock
+  class TokenGenerator < Padlock::Base
+    def initialize(params = {})
+      @user       = params[:user]
+      @type       = params[:type] || :auth
+      @expired_at = params[:expired_at] || Time.zone.now + 1.month
```

Comment:
use `fetch` when you set default values in initializer.

But if you need to wrap cases like this:
```
Object.new(type: nil)
```
and have `:auth` instead of `nil`
=> use your approach